plugins/main/bridge/bridge.go

```diff
@@ -260,12 +284,20 @@ func cmdAdd(args *skel.CmdArgs) error {
 
 	result.Interfaces = []*current.Interface{brInterface, hostInterface, containerInterface}
 
+	var firstV4Addr net.IP
+	var firstV6Addr net.IP
 	for _, ipc := range result.IPs {
 		// All IPs currently refer to the container interface
 		ipc.Interface = 2
 		if ipc.Gateway == nil && n.IsGW {
 			ipc.Gateway = calcGatewayIP(&ipc.Address)
 		}
+		switch {
+		case firstV4Addr == nil && ipc.Address.IP.To4() != nil:
+			firstV4Addr = ipc.Address.IP
+		case firstV6Addr == nil && len(ipc.Address.IP) == net.IPv6len:
```

Comment:
I'm not sure you can use length as a check here, since a V4 address can be 16 bytes long in Go too, hence the need for To4() everywhere.  Perhaps grab and use canonicalizeIP() from plugins/ipam/host-local/backend/allocator/allocator.go and then you can be sure?